### PR TITLE
[ADD] udes_stock: Warning message when mass (un)archiving

### PR DESCRIPTION
**Pull request by armandcela**
Story: 1403
Signed-off-by: Armand Cela <armand.cela@unipart.io>

### CODE REVIEW

File: addons/udes_stock/static/src/js/list_controller.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        var self = this;
+        function archiveThem() {
+            return self._archive(self.selectedRecords, archive);
+        }
+        if (archive) {
+            Dialog.confirm(this, _t("Are you sure you want to archive these records ? They will become unavailable and hidden from searches."), {
+                confirm_callback: archiveThem,
+            });
+        } else {
+            Dialog.confirm(this, _t("Are you sure you want to unarchive these records ?"), {
```
Comment (PeterAlabaster) [inline addons/udes_stock/static/src/js/list_controller.js:23]: super nit, not sure if there should be a space before the `?`

**Review by PeterAlabaster — APPROVED** [review]
lgtm one tiny comment

File: addons/udes_stock/static/src/js/list_controller.js (hunk 1 above)
Comment (armandcela) [reply addons/udes_stock/static/src/js/list_controller.js:23]: In your suggested help in delete records, there was a space before ? :)

File: addons/udes_stock/static/src/js/list_controller.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+var ListController = require('web.ListController');
+var Dialog = require('web.Dialog');
+var core = require('web.core');
+
+var _t = core._t;
+
+ListController.include({
+
+    _onToggleArchiveState: function (archive) {
+        //Override the method in order to add a dialog confirm before archiving or unarchiving the selected records
```
Comment (caleb-shelton) [inline addons/udes_stock/static/src/js/list_controller.js:13]: nit typo: a confirm dialog

File: addons/udes_stock/static/src/js/list_controller.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        var self = this;
+        function archiveThem() {
+            return self._archive(self.selectedRecords, archive);
+        }
+        if (archive) {
+            Dialog.confirm(this, _t("Are you sure you want to archive these records ? They will become unavailable and hidden from searches."), {
```
Comment (caleb-shelton) [inline addons/udes_stock/static/src/js/list_controller.js:19]: super nit typo again (sorry): records? (no space)

File: addons/udes_stock/static/src/js/list_controller.js (hunk 1 above)
Comment (PeterAlabaster) [reply addons/udes_stock/static/src/js/list_controller.js:23]: odoo don't have the best warning messages :)

Comment (armandcela) [reply addons/udes_stock/static/src/js/list_controller.js:23]: Convinced me now :)

**Review by caleb-shelton — APPROVED** [review]
LGTM

**Review by LoKKeR96 — APPROVED** [review]

**Review by adamrpatrick — APPROVED** [review]

**Pull request merged**